bin/varnishd/http2/cache_http2_deliver.c

```diff
@@ -92,6 +92,44 @@ h2_bytes(struct req *req, enum vdp_action act, void **priv,
 	return (0);
 }
 
+static struct h2_simple_response_msg {
+	const char	*p;
+	size_t		l;
+	unsigned	s;
+	const char	*r;
+} h2_simple_response_msg[] = {
+	[R_400] = { "\x92", 1, 400, "Bad Request"},
+#define SRESP(s, r) [R_##s] = { "\x18\x03" #s, 5, s, r}
+	SRESP(100, "Continue"),
+	SRESP(417, "Expectation Failed")
+};
```

Comment:
This part, I don't understand. This `SRESP` doesn't have to be the same as the H/1 version, you can add the `"\x??"` prefixes in the macro signature. You can also have `H1_SRESP` and `H2_SRESP` macros if you want to avoid confusion.

Also you forgot to `#undef` it here. Nitpick about the other `SRESP`, I would `#undef` it inside the curly braces.